repository/copier.go

```diff
@@ -135,25 +153,42 @@ func (c *HDFSCopier) CopyToRemote(src, dst string, localFs billy.Filesystem) (er
 	}
 
 	// TODO to avoid this, we should implement a 'truncate' flag in 'client.Create' method
-	_, err = c.client.Stat(p)
-	if err == nil {
-		err = c.client.Remove(p)
-		if err != nil {
-			return err
-		}
+	err = c.deleteIfExists(pCopy)
+	if err != nil {
+		return err
 	}
 
-	if err != nil && !os.IsNotExist(err) {
+	rf, err := c.client.Create(pCopy)
+	if err != nil {
 		return err
 	}
 
-	rf, err := c.client.Create(p)
+	// Delete temporary file in case the process is stopped while copying
+	defer func() {
+		rf.Close()
+		c.deleteIfExists(pCopy)
+	}()
+
+	_, err = io.Copy(rf, lf)
+
+	checkClose(rf, &err)
+
+	if err != nil {
+		c.client.Remove(pCopy)
+		return err
+	}
+
+	err = c.deleteIfExists(p)
```

Comment:
Now we can have the case that if pCopy was correctly generated, but the process stop on that line, then the next process that wants to add data to the siva file will delete the previous `.copy` file, and then all the info inside the siva files will disappear... :S

Reply:
You're right. There is a slight chance that is happens between the copy and the move. Still better that the big change while copying (could be several minutes).

I can not think of any atomic operation in hdfs or go but we could add more steps to be sure. What do you think about this?

* Copy to temporary file
* Move old file to `filename.old`
* Move temporary file to destination
* Delete `filename.old`

We may check also for the existence of `filename.old` to restore the data in case of failure

Reply:
This overcomplication is not needed as rename operation can substitute an already existing file.